x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -206,13 +207,19 @@ public void execute(
         );
         long start = configuration.getQueryStartTimeNanos();
         String local = RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY;
+        /*
+         * Grab the output attributes here, so we can pass them to
+         * the listener without holding on to a reference to the
+         * entire plan.
+         */
+        List<Attribute> outputAttributes = physicalPlan.output();
```

Comment:
Wow, very... Specific. How did you find this problem, or what did you see to think about this?

Reply:
A heap dump on out of memory errors caused by `testTooManyEval`. That has 20 * 490 `EvalOperator`s in it. If the plan isn't *huge* then it's generally not a big deal. But if you build something abusing like that and try to run it with as little memory as possible then this helps. Mostly it helps by cutting way down on the memory not tracked by the circuit breaker.

Reply:
Do you want to use the same pattern higher up around line 174 for the coordinator-only pathway? Perhaps that pathway is faster with a generally smaller plan, but might be better to have the same pattern in both places?

Reply:
:+1: